doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+
+Setup Data Tag
+^^^^^^^^^^^^^^
+
+The Linux Secure Launch protocol requires a Setup Indirect setup data entry to
+pass information to Secure Launch. This resulting structure is chained into
+Linux's `struct setup_header->setup_data` linked list.
+::
+
+    struct skl_tag_setup_data {
+        struct skl_tag_hdr hdr;
+        struct setup_data data;
+    } __packed;
+
+Logging Class
+-------------
+
+The logging class holds the tags related to the TPM Event log and SKL's
+internal logging facilities.
+
+Event Log Tag
+^^^^^^^^^^^^^
+
+The Event Log tag provides three controls, policy, scheme, and minimum size.
+The policy controls how and what algorithms will be used, scheme allows for
+selecting the PCR usage scheme, and minimum size sets the minimum amount of Log
+Buffer space that will be used for the Event Log, see Log Buffer section for
+more details.
+::
+
+    struct skl_tag_evtlog {  
+        struct skl_tag_hdr hdr;
+        u16 policy;
+        u16 scheme;
+        u32 min_size;       
+    } __packed;             
+
+CRTM Hash Tag
+^^^^^^^^^^^^^
+
+The SKINIT instruction does not provide the measurement it made of the SKL but
+the value needs to be present in the Event Log for attestation purposes. This
+tag provides the option for the DRTM Preamble to hash the SKL and provide the
+measurement to the SKL for recording into the log, see CRTM Log Event for more
+details.
+::
+
+    struct skl_tag_crtm_hash {
+        struct skl_tag_hdr hdr;
+        u16 algo_id;
+        u8 digest[];
+    } __packed;
+
+Kernel Log Tag
+^^^^^^^^^^^^^^
+
+The Kernel Log tag provides a means to control/configure logging done by SKL
+itself. Specifically to enable/disable and set priority of the logging to be
+done by SKL.
+::
+
+    struct skl_tag_crtm_hash {
+        struct skl_tag_hdr hdr;
+        u8 enable;
+        u8 priority;
+    } __packed;
+
+MSB Class
+---------
+
+The Measured Secure Boot (MSB) is an upcoming optional operational mode of the
+SKL. It allows for the SKL to function as the Root of Trust behind a signing
+key which can be used to start a SecureBoot RTV trust chain. The MSB Class is
+for all tags relating to configuring MSB.
+
+MSB Key Tag
+^^^^^^^^^^^
+
+The MSK Key tag is to allow embedding an MSB key directly into SKL.
+::
+
+    struct skl_tag_msb_key {
+        struct skl_tag_hdr hdr;
+        u16 algo_id;
+        u8 key[];
+    } __packed;
+
+MSB Key DB Tag
+^^^^^^^^^^^^^^
+
+The MSK Key DB tag points to a key db list external to the SKL along with a
+hash of the db list.
+::
+
+    struct skl_tag_msb_keydb {
+        struct skl_tag_hdr hdr;
+        u32 keydb;
+        u32 size;
+        u16 algo_id;
+        u8 digest[];
+    } __packed;
+
+IOMMU Table
+===========
+
+The IOMMU Table will be constructed at the first 4k page-aligned address after
+the Boot Tags but no further than 61440 bytes into the SKL. 
+
+Log Buffer
+==========
+
+The Log Buffer region contains the TPM Event Log and optionally a kernel log
```

Comment:
Currently for TPM event log we search for DRTM ACPI table that should have pointer and size of event log, and if not found, GRUB allocates memory for it and passes the tag, so it should be always available. Why do we want to change this logic?

Reply:
My mistake, we do want to keep that.